### ISSUE TITLE
`parse_structure` performance can be improved by less regular expressions

### ISSUE THREAD

**Issue by Hayashi-Yudai** [dcmoura/spyql#85]
## TL;DR

I found that `parse_structure` function uses regular expression. I think this function has not to use a lot of regular expression, and can be clearer and faster by refactoring.

https://github.com/dcmoura/spyql/blob/ca5b414d2ce0bfc16fcacfca94bb6a845c1b8724/spyql/parser.py#L59-L108

## Details

The code analyze queries the following steps,

1. For each keywords (eg. select, group by), search start index and end index from the query
2. Tokens between end index of i-th keyword and start index of (i+1)-th keyword is the argument of the keyword
3. Raise error when other keyword is in the element

In my opinion, this function can also be realized by

1. Split query by spaces or new lines into list (eg. "select * from [1,2,3]" -> ["select", "\*", "from", "[1,2,3]"])
2. Elements between keywords are arguments of keywords appear before them (eg.  ["select", "\*", "from", "[1,2,3]"] -> {"select": "*", "from": "[1,2,3]"})

This logic is clearer than present one. In addition, it improve performance because it does not use regular expression. (In my environment pytest execution time was reduced from 1.8 to 1.6 seconds on five occasions.)

If you are interested in this change, I would like to submit pull request.

**Comment by dcmoura** [comment]
Hey @Hayashi-Yudai! Good to hear from you again. Every contribution that makes the code cleaner and/or increases performance is welcomed :-) 

To be honest, when thinking on performance I tend to focus on the data processing part, and not so much on the query parsing since I expect it to be a small overhead, but I might be wrong. In which concerns query parsing, IMO, the criteria should be robustness first, followed by cleaner logic, followed by performance. 

Please pay attention to cases like `SELECT col1,col1*2 FROM range(0, 10)`, where the two output columns have not spaces in between, and the `FROM` clause has spaces between arguments.

**Comment by dcmoura** [comment]
Also @Hayashi-Yudai , if you are playing with parsing, it might make sense looking into #84
